source/as-promise.ts

```diff
@@ -3,30 +3,35 @@ import EventEmitter from 'events';
 import getStream from 'get-stream';
 import is from '@sindresorhus/is';
 import PCancelable from 'p-cancelable';
-import requestAsEventEmitter from './request-as-event-emitter';
-import {HTTPError, ParseError, ReadError} from './errors';
+import {NormalizedOptions, Response, CancelableRequest} from './utils/types';
 import {mergeOptions} from './merge';
+import {ParseError, ReadError, HTTPError} from './errors';
 import {reNormalizeArguments} from './normalize-arguments';
-import {CancelableRequest, Options, Response} from './utils/types';
+import requestAsEventEmitter from './request-as-event-emitter';
 
-export default function asPromise(options: Options) {
+export default function asPromise(options: NormalizedOptions): CancelableRequest<Response> {
 	const proxy = new EventEmitter();
 
-	const parseBody = (response: Response) => {
-		if (options.responseType === 'json') {
-			response.body = JSON.parse(response.body as string);
-		} else if (options.responseType === 'buffer') {
-			response.body = Buffer.from(response.body as Buffer);
-		} else if (options.responseType !== 'text' && !is.falsy(options.responseType)) {
-			throw new Error(`Failed to parse body of type '${options.responseType}'`);
+	const parseBody = (response: Response): void => {
+		switch (options.responseType) {
+			case 'json':
+				response.body = JSON.parse(response.body);
+				break;
+			case 'buffer':
+				response.body = Buffer.from(response.body);
+				break;
```

Comment:
Let's use the `if/else` chain here. It's very easy to forget about `break;`. It's not a good practice when you have many possible choices.